<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mediaplayerClass</class>
 <widget class="QMainWindow" name="mediaplayerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>871</width>
    <height>589</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>mediaplayer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>9</y>
      <width>851</width>
      <height>441</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="videowidget" name="openGLWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>450</y>
      <width>851</width>
      <height>61</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="vL1" stretch="0,0">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="hL2">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="playtime">
         <property name="text">
          <string>TextLabel</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="total_time">
         <property name="text">
          <string>TextLabel</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="XSlider" name="time_Slider">
         <property name="maximum">
          <number>999</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="hL1">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="playbtn">
         <property name="text">
          <string>播放</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pausebtn">
         <property name="text">
          <string>暂停</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="stopbtn">
         <property name="text">
          <string>停止</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="openbtn">
         <property name="text">
          <string>打开</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>871</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionffmpeg">
   <property name="text">
    <string>ffmpeg</string>
   </property>
  </action>
  <action name="actionopen">
   <property name="text">
    <string>open</string>
   </property>
  </action>
  <action name="actionsave">
   <property name="text">
    <string>save</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>videowidget</class>
   <extends>QOpenGLWidget</extends>
   <header>videowidget.h</header>
   <slots>
    <signal>clicked()</signal>
    <slot>play()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>XSlider</class>
   <extends>QSlider</extends>
   <header>xslider.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="mediaplayer.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>playbtn</sender>
   <signal>clicked()</signal>
   <receiver>mediaplayerClass</receiver>
   <slot>play()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>184</x>
     <y>531</y>
    </hint>
    <hint type="destinationlabel">
     <x>181</x>
     <y>550</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openbtn</sender>
   <signal>clicked()</signal>
   <receiver>mediaplayerClass</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>740</x>
     <y>531</y>
    </hint>
    <hint type="destinationlabel">
     <x>734</x>
     <y>552</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>time_Slider</sender>
   <signal>sliderPressed()</signal>
   <receiver>mediaplayerClass</receiver>
   <slot>sliderPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>502</y>
    </hint>
    <hint type="destinationlabel">
     <x>394</x>
     <y>561</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>time_Slider</sender>
   <signal>sliderReleased()</signal>
   <receiver>mediaplayerClass</receiver>
   <slot>sliderReleased()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>526</x>
     <y>508</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>560</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>play()</slot>
  <slot>open()</slot>
  <slot>sliderPressed()</slot>
  <slot>sliderReleased()</slot>
 </slots>
</ui>
